### PR TITLE
Oren/group listener

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/groups/groupsService.js [hunk 1]
```diff
@@ -127,6 +134,48 @@
...
+                }
+
+                function _getGroupPath(){
+                    return _getStorage().variables.appUserSpacePath + '/groups';
+                }
+
+                function addGroupListener() {
+                    var authData = _getAuthSrv().getAuth();
+                    if (authData && authData.uid) {
+                        var ref = new FirebaseListenerRef(authData.uid);
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: are you sure this is the right path (/ref) in firebase?

File: src/components/groups/groupsService.js [hunk 2]
```diff
@@ -127,6 +134,48 @@
...
+                        ref.on('child_added', groupsChildAdded);
+                        ref.on('child_removed', groupsChildRemoved);
+                    }
+                }
+
+                function groupsChildAdded(dataSnapshot) {
+                    $timeout(function () {
+                        console.log('GroupsService.groups', GroupsService.groups);
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: $log....why you print here groups ?

File: src/components/groups/groupsService.js [hunk 3]
```diff
@@ -127,6 +134,48 @@
...
+                        GroupsService.groups[dataSnapshot.key()] = dataSnapshot.val();
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: ?

File: src/components/groups/groupsService.js [hunk 4]
```diff
@@ -127,6 +134,48 @@
...
+                    });
+                }
+
+                function groupsChildRemoved(dataSnapshot) {
+                    $timeout(function () {
+                        delete GroupsService.groups[dataSnapshot.key()];
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: Does it also delete from firebase ?

File: src/components/groups/groupsService.js [hunk 5]
```diff
@@ -127,6 +134,48 @@
...
+                    });
+                }
+
+                function FirebaseListenerRef(uid) {
```
Comment (assafshp) [inline src/components/groups/groupsService.js]: function name is not good - too global

File: src/components/groups/groupsService.js (hunk 1 above)
Comment (orenor10) [inline src/components/groups/groupsService.js]: yes, this is the path to teacher groups

File: src/components/groups/groupsService.js (hunk 2 above)
Comment (orenor10) [inline src/components/groups/groupsService.js]: no need, I delete it

File: src/components/groups/groupsService.js (hunk 3 above)
Comment (orenor10) [inline src/components/groups/groupsService.js]: init the groups from the listener callback.

File: src/components/groups/groupsService.js (hunk 5 above)
Comment (orenor10) [inline src/components/groups/groupsService.js]: done

**Pull request merged**